src/bam/mod.rs

```diff
@@ -31,18 +31,18 @@ pub use crate::bam::header::Header;
 pub use crate::bam::record::Record;
 
 /// Implementation for `Read::set_threads` and `Writer::set_threads`.
-pub fn set_threads(htsfile: *mut htslib::htsFile, n_threads: usize) -> Result<()> {
+pub unsafe fn set_threads(htsfile: *mut htslib::htsFile, n_threads: usize) -> Result<()> {
```

Comment:
Did clippy suggest to move the unsafe here? I don't understand the rationale behind that. Do you have a clue?